### PR TITLE
api/nfd: document all undocumented fields in the types

### PR DESCRIPTION
**Pull request by marquiz**

### CODE REVIEW

File: api/nfd/v1alpha1/types.go [hunk 1]
```diff
@@ -141,12 +149,17 @@ type NodeFeatureGroup struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   NodeFeatureGroupSpec   `json:"spec"`
+	// Spec defines the rules to be evaluated.
+	Spec NodeFeatureGroupSpec `json:"spec"`
+
+	// Status of the node feature group after the most recent evaluation of the
```
Comment (ArangoGutierrez) [inline api/nfd/v1alpha1/types.go:155]:
```suggestion
	// Status of the nodeFeatureGroup after the most recent evaluation of the
```
no?

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
just one nit

File: api/nfd/v1alpha1/types.go (hunk 1 above)
Comment (marquiz) [reply api/nfd/v1alpha1/types.go]: I just spelled out the natural language version of the purpose of the object 🙂 But I don't have strong opinions. Changed as suggested.

Comment (ArangoGutierrez) [reply api/nfd/v1alpha1/types.go]: for node feature we have NodeFeature, so is just to have a similar form

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**